x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -207,6 +235,17 @@ public void executeOptimizedPlan(
         ActionListener<Result> listener
     ) {
         PhysicalPlan physicalPlan = logicalPlanToPhysicalPlan(optimizedPlan, request);
+        if (explainMode) {
```

Comment:
Should we pass these vars (explainMode, and the plan strings) as parameters instead? If I'm not wrong, calling session.execute() twice will leak those vars and break the second query. We're not doing that I think, but probably worth avoiding it (?)